### PR TITLE
Fix packaging to exclude logs and test reports

### PR DESCRIPTION
**Pull request by rjeschmi**
Also clean up some pylint complaints

### CODE REVIEW

File: easybuild/tools/package/utilities.py [hunk 1]
```diff
@@ -40,7 +40,8 @@
 from vsc.utils.missing import get_subclasses
 from vsc.utils.patterns import Singleton
 
-from easybuild.tools.config import PKG_TOOL_FPM, PKG_TYPE_RPM, build_option, get_package_naming_scheme
+from easybuild.tools.config import PKG_TOOL_FPM, PKG_TYPE_RPM, build_option, \
+    get_package_naming_scheme, log_path
```
Comment (boegel) [inline easybuild/tools/package/utilities.py]: no need to break the import line here? if it's less than 120 chars, keep it on the same line

File: easybuild/tools/package/utilities.py [hunk 2]
```diff
@@ -49,12 +50,13 @@
 from easybuild.tools.utilities import import_available_modules, quote_str
 
 
-_log = fancylogger.getLogger('tools.package')
+_LOG = fancylogger.getLogger('tools.package')
```
Comment (boegel) [inline easybuild/tools/package/utilities.py]: we usually don't use caps here, but it does make sense, so fine

Comment (boegel) [inline easybuild/tools/package/utilities.py]: it would be nicer to make this consistent across all modules though...

Comment (boegel) [inline easybuild/tools/package/utilities.py]: don't do that in this PR pls :)

File: easybuild/tools/package/utilities.py [hunk 3]
```diff
@@ -102,14 +104,20 @@ def package_with_fpm(easyblock):
...
-        _log.debug("The dep added looks like %s ", dep)
+        _LOG.debug("The dep added looks like %s ", dep)
         dep_pkgname = package_naming_scheme.name(dep)
         depstring += " --depends %s" % quote_str(dep_pkgname)
 
+    exclude_files_glob = [
+        os.path.join(log_path(), "*.log"),
+        os.path.join(log_path(), "*.md"),
+    ]
+    exclude_files_glob = [os.path.join(easyblock.installdir[1:], x) for x in exclude_files_glob]
```
Comment (boegel) [inline easybuild/tools/package/utilities.py]: what's the `[1:]` about?

File: easybuild/tools/package/utilities.py [hunk 4]
```diff
@@ -102,14 +104,20 @@ def package_with_fpm(easyblock):
...
     depstring = ''
     for dep in deps:
-        _log.debug("The dep added looks like %s ", dep)
+        _LOG.debug("The dep added looks like %s ", dep)
         dep_pkgname = package_naming_scheme.name(dep)
         depstring += " --depends %s" % quote_str(dep_pkgname)
 
+    exclude_files_glob = [
+        os.path.join(log_path(), "*.log"),
+        os.path.join(log_path(), "*.md"),
```
Comment (boegel) [inline easybuild/tools/package/utilities.py]: add comments here to make it clear that these are (EasyBuild) logs and test report

File: easybuild/tools/package/utilities.py [hunk 5]
```diff
@@ -121,15 +129,26 @@ def package_with_fpm(easyblock):
         '--iteration', pkgrel,
         '--description', quote_str(easyblock.cfg["description"]),
         '--url', quote_str(easyblock.cfg["homepage"]),
+    ]
+    excludes = ['--exclude %s' % quote_str(x) for x in exclude_files_glob]
```
Comment (boegel) [inline easybuild/tools/package/utilities.py]: collapse this with the 2nd `exclude_files_glob =`:

```python
excludes = ['--exclude %s' % quote_str(os.path.join(self.installdir[1:], x)) for x in exclude_files_glob]
```

Still not sure what the `[1:]` is about

File: easybuild/tools/package/utilities.py [hunk 6]
```diff
@@ -140,19 +159,20 @@ def package_with_fpm(easyblock):
...
         # rpmbuild is required for generating RPMs with FPM
         if pkgtool == PKG_TOOL_FPM and build_option('package_type') == PKG_TYPE_RPM:
             rpmbuild_path = which('rpmbuild')
             if rpmbuild_path:
-                _log.info("Required tool 'rpmbuild' found at %s", rpmbuild_path)
+                _LOG.info("Required tool 'rpmbuild' found at %s", rpmbuild_path)
             else:
-                raise EasyBuildError("rpmbuild is required when generating RPM packages with FPM, but was not found")
+                raise EasyBuildError("rpmbuild is required when generating RPM "
+                                     "packages but was not found")
```
Comment (boegel) [inline easybuild/tools/package/utilities.py]: don't wrap the string across two lines, it's < 120 chars?

File: easybuild/tools/package/utilities.py [hunk 7]
```diff
@@ -173,20 +193,20 @@ def __init__(self):
         if sel_pns in avail_pns:
             self.pns = avail_pns[sel_pns]()
         else:
-            raise EasyBuildError("Selected package naming scheme %s could not be found in %s",
-                                 sel_pns, avail_pns.keys())
+            raise EasyBuildError("Selected package naming scheme %s could not "
+                                 "be found in %s", sel_pns, avail_pns.keys())
```
Comment (boegel) [inline easybuild/tools/package/utilities.py]: same here, why wrap?

if you need to wrap, don't break the string:

```python
raise EasyBuildError("Selected package naming scheme %s could not be found in %s",
                     sel_pns, avail_pns.keys())
```

File: easybuild/tools/package/utilities.py [hunk 8]
```diff
@@ -173,20 +193,20 @@ def __init__(self):
...
 
-    def version(self, ec):
+    def version(self, easyconfig):
         """Determine package version"""
-        version = self.pns.version(ec)
+        version = self.pns.version(easyconfig)
         return version
 
-    def release(self, ec):
+    def release(self, easyconfig):  # pylint: disable=W0613
```
Comment (boegel) [inline easybuild/tools/package/utilities.py]: what are you disabling here? what doesn't pylint like about this one?

File: easybuild/tools/package/utilities.py (hunk 7 above)
Comment (rjeschmi) [inline easybuild/tools/package/utilities.py]: It was a line length issue, both pep8 and pylint complain about it. 

If we want to make a standard longer line I'm fine with that.

Comment (boegel) [inline easybuild/tools/package/utilities.py]: our code style allows lines up until 120 chars, the default in pep8/pylint is 80 I think (which is a nightmare imho)

File: easybuild/tools/package/utilities.py (hunk 8 above)
Comment (rjeschmi) [inline easybuild/tools/package/utilities.py]: I don't use the variable easyconfig, but it is probably useful to include it as a sort of documentation.

Comment (boegel) [inline easybuild/tools/package/utilities.py]: shouldn't it be passed down to `self.pns.release` (which can then choose to ignore it)?

**Comment by boegel** [comment]
Jenkins: ok to test

File: easybuild/tools/package/utilities.py [hunk 9]
```diff
@@ -110,6 +111,18 @@ def package_with_fpm(easyblock):
...
         depstring += " --depends %s" % quote_str(dep_pkgname)
 
+    # Excluding the EasyBuild logs and test reports that might be in the installdir
+    exclude_files_glob = [
+        os.path.join(log_path(), "*.log"),
+        os.path.join(log_path(), "*.md"),
+    ]
+    # stripping off leading / to match expected glob in fpm
+    exclude_files_glob = [
+        '--exclude %s' % quote_str(os.path.join(easyblock.installdir[1:], x))
```
Comment (boegel) [inline easybuild/tools/package/utilities.py]: use `easyblock.installdir.lstrip(os.pathsep)`

you have no idea what you're stripping off with `[1:]`

File: easybuild/tools/package/utilities.py [hunk 10]
```diff
@@ -110,6 +111,18 @@ def package_with_fpm(easyblock):
...
+        for x
+        in exclude_files_glob
+    ]
```
Comment (boegel) [inline easybuild/tools/package/utilities.py]: no need to split across multiple lines? it looks very weird to me this way

File: easybuild/tools/package/utilities.py [hunk 11]
```diff
@@ -121,10 +134,20 @@ def package_with_fpm(easyblock):
...
         '--description', quote_str(easyblock.cfg["description"]),
         '--url', quote_str(easyblock.cfg["homepage"]),
+    ]
+    _log.debug("excludes list: %s", exclude_files_glob)
+    cmdlist.extend(exclude_files_glob)
+
+    if build_option('debug'):
+        cmdlist.extend([
+            '--debug',
+        ])
```
Comment (boegel) [inline easybuild/tools/package/utilities.py]: do this single line, with `.append`:

```python
cmdlist.append('--debug')
```

**Comment by boegel** [comment]
@rjeschmi: can we enhance the unit test(s) we have for the packaging support to check that logs/test report are *not* included?

File: test/framework/package.py [hunk 12]
```diff
@@ -195,11 +216,18 @@ def test_package(self):
         self.assertTrue(os.path.isfile(pkgfile), "Found %s" % pkgfile)
 
         pkgtxt = read_file(pkgfile)
-        pkgtxt_regex = re.compile("Contents of installdir %s" % easyblock.installdir)
+        pkgtxt_regex = re.compile("STARTCONTENTS of installdir %s" % easyblock.installdir)
         self.assertTrue(pkgtxt_regex.search(pkgtxt), "Pattern '%s' found in: %s" % (pkgtxt_regex.pattern, pkgtxt))
 
+        no_logfiles_regex = re.compile(r'STARTCONTENTS.*\.(log|md)$.*ENDCONTENTS', re.DOTALL|re.MULTILINE)
+        self.assertFalse(no_logfiles_regex.search(pkgtxt), "Pattern '%s' found in: %s" % (no_logfiles_regex.pattern, pkgtxt))
```
Comment (boegel) [inline test/framework/package.py]: this should say `"Pattern '%s' not found in: %s"`

File: easybuild/tools/package/utilities.py [hunk 13]
```diff
@@ -121,10 +133,18 @@ def package_with_fpm(easyblock):
         '--iteration', pkgrel,
         '--description', quote_str(easyblock.cfg["description"]),
         '--url', quote_str(easyblock.cfg["homepage"]),
+    ]
+    _log.debug("excludes list: %s", exclude_files_glob)
```
Comment (boegel) [inline easybuild/tools/package/utilities.py]: you're logging this twice?

File: easybuild/tools/package/utilities.py [hunk 14]
```diff
@@ -110,6 +111,17 @@ def package_with_fpm(easyblock):
...
+    exclude_files_glob = [
+        os.path.join(log_path(), "*.log"),
+        os.path.join(log_path(), "*.md"),
+    ]
+    # stripping off leading / to match expected glob in fpm
+    exclude_files_glob = [
+        '--exclude %s' % quote_str(os.path.join(easyblock.installdir.lstrip(os.sep), x))
+        for x in exclude_files_glob
+    ]
+    _log.debug("exclude_glob: %s", exclude_files_glob)
```
Comment (boegel) [inline easybuild/tools/package/utilities.py]: what's `exclude_glob`?

File: test/framework/package.py (hunk 12 above)
Comment (rjeschmi) [inline test/framework/package.py]: Well it is the error message, so the error is that the pattern is found, no?

Comment (boegel) [inline test/framework/package.py]: you'll get an error like:

```AssertionError: Pattern '...' found in ...`

The assertion you're making is that the pattern is *not* found, however, and that's what's being reported as an error.

You can interpret it both ways though, but we generally mention what *should* be there in the error message.

**Comment by boegel** [comment]
Going in, thanks @rjeschmi!

**Pull request merged**